decidim-core/lib/decidim/participatory_space_manifest.rb

```diff
@@ -50,5 +50,9 @@ def seed!
     def route_name
       super || model_class_name.demodulize.underscore
     end
+
+    def public_spaces(&block)
```

Comment:
Statistics are also used on the admin panel so this might not work there. What if we provided `participatory_spaces` and `features` and we used a `public` scope on top of them when needed? Spaces and Features are always gonna be `ActiveRecord` models so we can rely on that working in future cases.

Thoughts @mrcasals?

Reply:
Ok, after talking about it *offline*, this won't break existing functionality. But I still feel like this is a *leaky* abstraction: We're asking for a very specific subset of data. By returning *all* the participatory spaces, we could then use this to build other features on top of it, while returning only the *public* limits us in some ways.

So, what I suggest is:

1) Making the contract more generic and as for *participatory_spaces*.
2) Adding a *scope* called *public* in all participatory spaces.
3) Create a *shared example* for participatory spaces that acts as a linter so tests on spaces will fail if we change / extend this contract in the future.

@deivid-rodriguez @oriolgual @mrcasals thoughts?

Reply:
Without giving too much thought to it, what you suggest makes sense to me and it certainly makes the API more robust and stable.

Reply:
Yeah, +1 to me, but I'm out on holidays and I don't know when I'll have time to work on this :disappointed:

Reply:
No worries, this isn't an *urgent* bug, so you can happily work on it when you return xDDD